pkg/login/social/azuread_oauth.go

```diff
@@ -127,8 +148,64 @@ func hasRole(roles []string, role models.RoleType) bool {
 	return false
 }
 
-func extractGroups(claims azureClaims) []string {
-	groups := make([]string, 0)
-	groups = append(groups, claims.Groups...)
-	return groups
+type getAzureGroupRequest struct {
+	SecurityEnabledOnly bool `json:"securityEnabledOnly"`
+}
+
+type getAzureGroupResponse struct {
+	Value []string `json:"value"`
+}
+
+func extractGroups(client *http.Client, claims azureClaims, token *oauth2.Token) ([]string, error) {
+	if len(claims.Groups) > 0 {
+		return claims.Groups, nil
+	} else if claims.ClaimNames.Groups != "" {
+		// If user groups exceeds 200 no groups will be found in claims.
+		// See https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#groups-overage-claim
+		parsedToken, err := jwt.ParseSigned(token.AccessToken)
+		if err != nil {
+			return nil, errutil.Wrapf(err, "error parsing id token")
+		}
+
+		var accessClaims azureAccessClaims
+		if err := parsedToken.UnsafeClaimsWithoutVerification(&accessClaims); err != nil {
+			return nil, errutil.Wrapf(err, "error getting claims from access token")
+		}
+
+		// Sadly the endpoints provided in _claim_source is pointed to the deprecated "graph.windows.net" api
+		// See https://docs.microsoft.com/en-us/graph/migrate-azure-ad-graph-overview
+		// So instead we do the group lookup to microsoft graph api
```

Comment:
I would still use the endpoints in `_claim_source` if that still works. That's more resilient toward them deciding to drop this endpoint too, since that's following some kind of standard.

Reply:
Yeah, i just need to figure out what permissions is required to access that endpoint then. Got access denied error when trying to use it